src/pages/RFOX/components/Claim/Claim.tsx

```diff
@@ -45,89 +41,75 @@ const ClaimStatus = makeSuspenseful(
   defaultBoxSpinnerStyle,
 )
 
-const ClaimEntries = [ClaimRoutePaths.Select, ClaimRoutePaths.Confirm, ClaimRoutePaths.Status]
-
 export const Claim: React.FC<ClaimRouteProps> = ({ headerComponent, setStepIndex }) => {
-  return (
-    <MemoryRouter initialEntries={ClaimEntries} initialIndex={0}>
-      <ClaimRoutes headerComponent={headerComponent} setStepIndex={setStepIndex} />
-    </MemoryRouter>
-  )
+  return <ClaimRoutes headerComponent={headerComponent} setStepIndex={setStepIndex} />
 }
 
 export const ClaimRoutes: React.FC<ClaimRouteProps> = ({ headerComponent, setStepIndex }) => {
   const location = useLocation()
   const queryClient = useQueryClient()
 
-  const [confirmedQuote, setConfirmedQuote] = useState<RfoxClaimQuote | undefined>()
   const [claimTxid, setClaimTxid] = useState<string | undefined>()
 
-  const stakingAssetAccountAddress = useMemo(() => {
-    return confirmedQuote ? fromAccountId(confirmedQuote.stakingAssetAccountId).account : undefined
-  }, [confirmedQuote])
-
-  const { queryKey: unstakingRequestQueryKey } = useGetUnstakingRequestsQuery({
-    stakingAssetAccountAddress,
-  })
+  const selectedUnstakingRequest = location.state?.selectedUnstakingRequest as
```

Comment:
location.state as source of truth for claims to avoid weird bugs we had - we now `navigate()` with a `selectedUnstakingRequest` as location state, so no need for props drilling bs

Reply:
`Location` *does* support a generic, but `useLocation` doesn't make use of it :(

https://github.com/remix-run/react-router/blob/eb19c3fae9e72aee3cfafb78bc87cae64c3b725b/packages/react-router/lib/hooks.tsx#L129

See https://github.com/DefinitelyTyped/DefinitelyTyped/discussions/60159 - interestingly, `@types/react-router` and `@types/react-router-dom` *were* properly typed, though they only go as far as v5. v6 now ships types, but has this wrong typing for location so casting is our best bet unfortunately!